### PR TITLE
feat: add `reset()` to utils proxy

### PR DESCRIPTION
**Pull request by v1narth**
## 🎯 Changes

- Added `resetQueries` support to utils proxy.
- Added unit test 

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/next/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react-query/src/internals/context.tsx [hunk 1]
```diff
@@ -166,6 +168,26 @@ export interface TRPCContextState<
...
+  ): Promise<void>;
+
+  /**
+   * @link https://react-query.tanstack.com/reference/QueryClient#queryclientresetqueries
+   */
+  resetQueries(
+    filters?: ResetQueryFilters,
+    options?: ResetOptions,
+  ): Promise<void>;
+
```
Comment (sachinraja) [inline packages/react-query/src/internals/context.tsx:190]: I don't think we should support this. What do you think @KATT?

Comment (v1narth) [reply packages/react-query/src/internals/context.tsx:190]: Just had a place where we really needed this functionality, if there are any alternatives maybe we could use them, but could not found any :)

The functionality we was looking for was the ability to reset a query without an observer.

Comment (KATT) [reply packages/react-query/src/internals/context.tsx:190]: How come you don't think we should support it, @sachinraja?

Comment (sachinraja) [reply packages/react-query/src/internals/context.tsx:190]: I meant specifically for the v9, interop API. We should definitely add the ability to reset queries. Are we still going to add new features for the interop API?

**Comment by trymbill** [comment]
Was looking for exactly this helper method. Any idea when this PR could be merged?

File: packages/react-query/src/internals/context.tsx (hunk 1 above)
Comment (KATT) [reply packages/react-query/src/internals/context.tsx:190]: Nono, not for v9!

**Review by KATT — APPROVED** [review]

**Review by KATT — CHANGES_REQUESTED** [review]
Ooops. Sorry for the delay on this. Happy to merge it but it's gotten out of date now - could you have a look at it @v1narth?

**Comment by v1narth** [comment]
@KATT Should I approve the Vercel - next-prisma-starter deploment? Or is it done by you?

**Review by KATT — APPROVED** [review]

**Comment by KATT** [comment]
@v1narth if you setup github sponsors I'll happily send you a little thank you for your efforts! :)

**Comment by v1narth** [comment]
@KATT Done, and thank you! ☺️

**Pull request merged**